src/Domain/auth/AuthenticationDomain.cs

```diff
@@ -0,0 +1,91 @@
+using System.Text.RegularExpressions;
+using Domain.dao;
+using Domain.utils;
+using Dto;
+using Dto.tools;
+
+namespace Domain.auth;
+
+public class AuthenticationDomain : IAuthenticationDomain {
+
+    private readonly IAuthenticationDao _authDao;
+    public AuthenticationDomain(IAuthenticationDao authDao) {
+        this._authDao = authDao;
+    }
+
+    public async Task<UserDto> ValidateAdmin(LoginRequestDto loginRequest) {
+        CheckForValidEmail(loginRequest.Email);
+        CheckForValidPassword(loginRequest.Password);
+
+        UserDto? userFromDatabase = await _authDao.GetUserByEmail(loginRequest.Email);
+        if (userFromDatabase is null) {
+            throw new ExceptionWithErrorCode(ErrorCode.NotFound, ErrorMessages.EmailDoesntExist);
+        }
+
+        bool doesPasswordMatch = VerifyPassword(loginRequest.Password, userFromDatabase.Password);
```

Comment:
`        if (VerifyPassword(loginRequest.Password, userFromDatabase.Password))
        {
            return userFromDatabase;
        }
        else
        {
            throw new ExceptionWithErrorCode(ErrorCode.BadRequest, ErrorMessages.IncorrectPassword);
        }`

Reply:
I disagree. The current way is more clear.